src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java

```diff
@@ -443,15 +416,14 @@ public void execute()
         return modulepathElements;
     }
 
-    private String getExecutable()
-        throws MojoFailureException
+    private JLinkExecutor getExecutor() throws MojoFailureException
     {
-        String jLinkExec;
+        JLinkExecutor jLinkExec;
         try
         {
-            jLinkExec = getJLinkExecutable();
+            jLinkExec = getJlinkExecutor();
         }
-        catch ( IOException e )
+        catch ( RuntimeException e )
```

Comment:
can you be more specific here?

Reply:
Done. Moved this outdated try-catch to the execution.